saltgui/static/scripts/routes/templates.js

```diff
@@ -0,0 +1,85 @@
+class TemplatesRoute extends PageRoute {
+
+  constructor(router) {
+    super("^[\/]templates$", "Templates", "#page_templates", "#button_templates", router);
+    this.jobsLoaded = false;
+
+    this._updateTemplates = this._updateTemplates.bind(this);
+    this._applyTemplate = this._applyTemplate.bind(this);
+  }
+
+  onShow() {
+    const templates = this;
+    return new Promise(function(resolve, reject) {
+      templates.resolvePromise = resolve;
+      if(templates.jobsLoaded) resolve();
+      templates.router.api.getJobs().then(templates._updateJobs);
+      templates.router.api.getConfigValues().then(templates._updateTemplates);
+    });
+  }
+
+  _updateTemplates(data) {
+    const container = this.getPageElement().querySelector(".templates");
+    
+    // should we update it or just use from cache (see commandbox) ?
+    const templates = data.return[0].data.return.saltgui_templates;
+    window.localStorage.setItem("templates", JSON.stringify(templates));
+    const keys = Object.keys(templates).sort();
+    for(const key of keys) {
+      const template = templates[key];
+      this._addTemplate(container, key, template);
+    }
+  }
+
+  _addTemplate(container, name, template) {
+    const tr = document.createElement("tr");
+
+    tr.appendChild(Route._createTd("name", name));
+
+    // calculate description
+    const description = template["description"];
+    if (!description) {
+      tr.appendChild(Route._createTd("description none", "(none)"));
+    } else {
+      tr.appendChild(Route._createTd("description", description));
+    }
+
+    // calculate target
+    const target = template["target"];
+    if (!target) {
+      tr.appendChild(Route._createTd("target none", "(none)"));
+    } else {
+      tr.appendChild(Route._createTd("target", target));
+    }
+
+    // calculate targettype
+    let targettype = template["targettype"];
+    if (!targettype) targettype = "";
+    tr.appendChild(Route._createTd("targettype", targettype));
```

Comment:
I think the column "Target Type" can be removed from the screen.
When it has a value, it can simply be placed before the target value.

Reply:
This is what I would like to debate with you:). I think that it's useful to have this information id dedicated column as the aim of this is to show (exact) what we have under master configuration and here we have dedicated field `targettype`.

Reply:
Ok. I've merged this as you mentioned as it doesn't look so bad :)